autobahn/wamp/component.py

```diff
@@ -381,7 +396,7 @@ def do_subscription(session, details):
             return fn
         return decorator
 
-    def register(self, uri, options=None):
+    def register(self, uri, options=None, validate=False):
```

Comment:
That parameter could be used to toggle the behavior, making it optional per procedure and "off" by default

Reply:
`Component.register` should mirror the args to the "real" register .. so, this is fine for a PoC but `session.register` should probably also take `validate=` if we're going to integrate this fully. I think most users are probably using `Session.register(..)` right now.

Reply:
Fixed that now; moved the validate code to protocol.py.